Core/include/Acts/Utilities/MathHelpers.hpp

```diff
@@ -0,0 +1,30 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <cmath>
+
+namespace Acts {
+
+template <typename T>
+inline auto square(T x) {
+  return std::pow(x, 2);
```

Comment:
why `std::pow` instead of `x*x` ?

Reply:
No particular reason. I it should be equivalent

Reply:
from some local tests it looks like `std::pow` is considerably more expensive then a simple multiplication

Reply:
I think I discussed this once with @stephenswat and I believe a good optimizer should be able to substitute that. But you might be right in reducing the complexity by one step is more conservative. Will do

Reply:
We had the discussion also here, and I also came. to the conclusion, that it shouldn't matter for squaring.

https://github.com/acts-project/acts/pull/3292#discussion_r1738601927